instrumentation/xxl-job/xxl-job-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/xxljob/common/XxlJobExperimentalAttributeExtractor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.xxljob.common;
+
+import com.xxl.job.core.glue.GlueTypeEnum;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import javax.annotation.Nullable;
+
+public class XxlJobExperimentalAttributeExtractor
+    implements AttributesExtractor<XxlJobProcessRequest, Void> {
+
+  private static final AttributeKey<String> XXL_JOB_GLUE_TYPE =
+      AttributeKey.stringKey("scheduling.xxl-job.glue.type");
+
+  private static final AttributeKey<String> XXL_JOB_RESULT_STATUS =
```

Comment:
I think you should also test a failing job. In my opinion using `success` and `failure` would be better that `true` and `false`. Even better would be to delete this and use the `SpanStatusExtractor` to set the span status to `ERROR` when the job fails.

Reply:
> I think you should also test a failing job. In my opinion using `success` and `failure` would be better that `true` and `false`. Even better would be to delete this and use the `SpanStatusExtractor` to set the span status to `ERROR` when the job fails.

I used `SpanStatusExtractor` instead of previous way now and found there is a bug in setting span status in 2.3.0+. Because, it changed from a method with a return value to one without a return value. https://github.com/xuxueli/xxl-job/blob/761de38a0b2a39706e2008e7914fba13bf4ca184/xxl-job-core/src/main/java/com/xxl/job/core/handler/IJobHandler.java#L16 So getting result code need to adjusted by a new class [XxlJobContext](https://github.com/xuxueli/xxl-job/blob/761de38a0b2a39706e2008e7914fba13bf4ca184/xxl-job-core/src/main/java/com/xxl/job/core/context/XxlJobContext.java#L98) added in 2.3.0. Then, I added a new module of xxl-job-2.3.0 to solve it. Above logic code you can see [instrumentation/xxl-job/xxl-job-2.3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/xxljob/v2_3_0/XxlJobHelper.java](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10421/files#diff-9112207319b785f61ed0870d44ccb3a7fc101ca0d5f65dfb812dfd6c216bc77f)